Code/Tools/AssetProcessor/native/AssetManager/assetScannerWorker.cpp

```diff
@@ -101,15 +108,14 @@ void AssetScannerWorker::ScanForSourceFiles(const ScanFolderInfo& scanFolderInfo
         AZ::u64 fileSize = isDirectory ? 0 : entry.size();
         AssetFileInfo assetFileInfo(absPath, modTime, fileSize, &rootScanFolder, isDirectory);
 
-        // Filtering out excluded files
-        if (m_platformConfiguration->IsFileExcluded(absPath))
-        {
-            m_excludedList.insert(AZStd::move(assetFileInfo));
-            continue;
-        }
-
         if (isDirectory)
         {
+            // Filtering out excluded directories immediately (not in a thread pool) since that prevents us from recursing.
+            if (m_platformConfiguration->IsFileExcluded(absPath))
+            {
+                m_excludedList.insert(AZStd::move(assetFileInfo));
```

Comment:
So now m_excludedList is just an excluded list of folders, right? 

Or should should the m_excludedList be filled out with the file entries during the blockingFilteredReduced() ?

Reply:
Seems like m_excludedList is a list of file entries (AssetFileInfo) instead of folders here.

Reply:
thats a really good question.  I didn't add m_excludedList, it was there before, but _whats it for???_

Reply:
It looks like the excluded list is only used to add information about excluded files into the file system cache (the one you ask about modtimes and stuff).  But why?

Reply:
I don't think we should be sending the excluded files into the file system cache at all.  I'll investigate some more.

Reply:
So it seems that the FileStateCache uses the file info for API such as FileStateCache::Exists() and FileStateCache::GetHash() so maybe the excluded files info is valid?

Reply:
I've kept the exclusion list working as it did before, essentially.  We can examine getting rid of it later, but I did other optimizations such that it doesn't actually matter anymore in terms of time taken.